tests/rkt_trust_test.go

```diff
@@ -0,0 +1,160 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
+)
+
+func runImage(t *testing.T, ctx *rktRunCtx, imageFile string, expected string, shouldFail bool) {
+	cmd := fmt.Sprintf(`%s --debug run --mds-register=false %s`,
+		ctx.cmd(), imageFile)
+	t.Logf("%s\n", cmd)
+	child, err := gexpect.Spawn(cmd)
+	if err != nil {
+		t.Fatalf("Cannot exec rkt")
+	}
+
+	if err = expectWithOutput(child, expected); err != nil {
+		t.Fatalf("Expected but didn't find %q in %v", expected, err)
+	}
+
+	err = child.Wait()
+	const exit1 = "exit status 1"
+	switch {
+	case shouldFail && err == nil:
+		t.Fatalf("Expected test to fail but it didn't")
+	case !shouldFail && err != nil:
+		t.Fatalf("rkt didn't terminate correctly: %v", err)
+	case err != nil && err.Error() != exit1:
+		t.Fatalf("rkt terminated with unexpected error: %v", err)
+	default:
+	}
+
+}
+
+func runRktTrust(t *testing.T, ctx *rktRunCtx, prefix string) {
+	var cmd string
+	if prefix == "" {
+		cmd = fmt.Sprintf(`%s trust --root %s`, ctx.cmd(), "key.gpg")
+	} else {
+		cmd = fmt.Sprintf(`%s trust --prefix %s %s`, ctx.cmd(), prefix, "key.gpg")
+	}
+
+	t.Logf("%s\n", cmd)
+	child, err := gexpect.Spawn(cmd)
+	if err != nil {
+		t.Fatalf("Cannot exec rkt trust: %s", err)
+	}
+
+	expected := "Are you sure you want to trust this key"
+	if err = expectWithOutput(child, expected); err != nil {
+		t.Fatalf("Expected but didn't find %q in %v", expected, err)
+	}
+
+	if err = child.SendLine("yes"); err != nil {
+		t.Fatalf("Cannot confirm rkt trust: %s", err)
+	}
+
+	if prefix == "" {
+		expected = "Added root key at"
+	} else {
+		expected = fmt.Sprintf(`Added key for prefix "%s" at`, prefix)
+	}
+	if err = expectWithOutput(child, expected); err != nil {
+		t.Fatalf("Expected but didn't find %q in %v", expected, err)
+	}
+
+	err = child.Wait()
+	if err != nil {
+		t.Fatalf("rkt trust didn't terminate as expected: %v", err)
+	}
+}
+
+func runSignImage(t *testing.T, ctx *rktRunCtx, imageFile string) {
+	dir, err := os.Getwd()
+	if err != nil {
+		t.Fatalf("Cannot get current working directory")
+	}
+
+	cmd := fmt.Sprintf("gpg --no-default-keyring --secret-keyring %s/secring.gpg --keyring %s/pubring.gpg --default-key D9DCEF41 --output %s.asc --detach-sig %s",
+		dir, dir, imageFile, imageFile)
+	t.Logf("%s\n", cmd)
+	child, err := gexpect.Spawn(cmd)
+	if err != nil {
+		t.Fatalf("Cannot exec gpg: %s", err)
+	}
+
+	err = child.Wait()
+	if err != nil {
+		t.Fatalf("gpg terminate as expected: %v", err)
+	}
+}
+
+func TestTrust(t *testing.T) {
+	imageFile := patchTestACI("rkt-inspect-trust1.aci", "--exec=/inspect --print-msg=Hello", "--name=rkt-prefix.com/my-app")
+	defer os.Remove(imageFile)
+	defer os.Remove(imageFile + ".asc")
```

Comment:
It probably doesn't matter but the signature file is not created until you call `runSignImage` so this should appear after that.